movie-browser/src/App.js

```diff
@@ -3,10 +3,14 @@ import Header from './components/Header.js';
 import Featured from './components/Featured.js';
 import Movies from './components/Movies.js';
 import Footer from './components/Footer.js';
+import MostViewed from './components/MostViewed.js';
 //import fetchMovies from './components/Api.js';
 
 class App extends Component {
-  state = { movies: [] };
+  state = { 
+    movies: [],
+    shouldShowAllMovies: false
+   };
 
   componentDidMount() {
     fetch('https://ghibliapi.herokuapp.com/films')
```

Comment:
u still need to update the state after this fetch

Reply:
Isn't that what I do with this?:
`    .then(response => response.json())
    .then(data => this.setState({ movies: data }));
  }`

Reply:
I just saw API.js. Two things, first, it is not a component so it shouldn't be on the components folder. Second, u won't `setState` because it is not a react component, it is just a helper/util function. You just need to return the data, not setting state. The setting of the state will be done on App.js

Reply:
Do you mean strip the Api.js down to this:
```const URL = 'https://ghibliapi.herokuapp.com/';

const fetchMovies = param => {
   fetch(`${URL}${param}`)
        .then(response => response.json())
        }

export default fetchMovies;
`
then finish the call in App like this:
``` componentDidMount() {
    fetchMovies('films')
    .then(data => this.setState({ movies: data }));
  }`

Reply:
correct!

Reply:
I'm going to merge but would like to continue the discussion about this on Slack